src/dynamo-streams.ts

```diff
@@ -170,82 +196,98 @@ export class DynamoStreamHandler<Entity, Context> {
 
       context.logger.info({ event }, 'Processing DynamoDB stream event');
 
-      // Iterate through every event.
-      for (const record of event.Records) {
-        const recordLogger = this.config.logger.child({ record });
-        if (!record.dynamodb) {
-          recordLogger.error(
-            { record },
-            'The dynamodb property was not present on event',
-          );
-          continue;
-        }
-
-        // Unmarshall the entities.
-        const oldEntity =
-          record.dynamodb.OldImage &&
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-          this.config.parse(unmarshall(record.dynamodb.OldImage as any));
-
-        const newEntity =
-          record.dynamodb.NewImage &&
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-          this.config.parse(unmarshall(record.dynamodb.NewImage as any));
-
-        // Handle INSERT events -- invoke the INSERT actions in order.
-        if (record.eventName === 'INSERT') {
-          if (!newEntity) {
-            recordLogger.error(
-              { record },
-              'No NewImage was defined for an INSERT event',
-            );
-            continue;
-          }
-
-          for (const action of this.actions.insert) {
-            await action({ ...context, logger: recordLogger }, newEntity);
-          }
-        }
-        // Handle MODIFY events -- invoke the MODIFY actions in order.
-        else if (record.eventName === 'MODIFY') {
-          if (!oldEntity) {
-            recordLogger.error(
-              { record },
-              'No OldImage was defined for a MODIFY event',
+      await processWithOrdering(
+        {
+          items: event.Records,
+          orderBy: (record) => {
+            // This scenario should only ever happen in tests.
+            if (!record.dynamodb?.Keys) {
+              return uuid();
+            }
+            return deterministicStringify(
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+              unmarshall(record.dynamodb.Keys as any),
```

Comment:
The key doesn't need to be unmarshalled here.

Reply:
Hmm...what makes you say that? I'm certain the value is marshalled. If you do a sumo search for `Processing DynamoDB stream event` and examine an example event, you will see this:

<img width="423" alt="Screen Shot 2023-08-02 at 11 56 03 AM" src="https://github.com/lifeomic/delta/assets/14932834/215c14dc-efea-4958-8a4a-af45f914dab4">

Are you saying that we could just stringify the marshalled key? If so, I think that would work. But, it seems a little clearer to me to just unmarshall the value -- esp. for a developer debugging the value.

Curious what other reviewers think.

Reply:
Because you are only using the stringified value as a sort key, it doesn't matter what the contents are, just that keys are in the same order.  Putting in the unmarshall is just another point of failure, and extra code.

Reply:
Yep, I'm convinced. Updating, and going to inline the deterministicStringify behaviro.